src/main/java/seedu/address/logic/commands/TagCommand.java

```diff
@@ -97,9 +131,34 @@ public CommandResult addUserTags(Model model) throws CommandException {
         model.getObservablePersonList().forEach(person ->
                 person.setCommonModules(editedUser.getImmutableModuleTags()));
 
+        for (ModuleTag tag : moduleTags) {
+            String day = tag.getDay();
+            String start = tag.getStartTime();
+            String end = tag.getEndTime();
+            if (day == null || start == null || end == null) {
+                continue;
+            }
+
+            Module mod = new Module(tag.toString());
+            int startHour = Integer.parseInt(start.substring(0, 2));
+            int startMin = Integer.parseInt(start.substring(2));
+            int endHour = Integer.parseInt(end.substring(0, 2));
+            int endMin = Integer.parseInt(end.substring(2));
+            LocalTime startTime = new LocalTime(startHour, startMin);
+            LocalTime endTime = new LocalTime(endHour, endMin);
```

Comment:
Please also try to abstract this out. You can do so in `TimeUtils`.

Reply:
Agreed, especially since you are repeatedly performing this operation.